lib/valkyrie/persistence/solr/model_converter.rb

```diff
@@ -5,20 +5,31 @@ module Valkyrie::Persistence::Solr
   class ModelConverter
     attr_reader :resource, :resource_factory
     delegate :resource_indexer, to: :resource_factory
+
+    # @param [Valkyrie::Resource] resource
+    # @param [ResourceFactory] resource_factory
     def initialize(resource, resource_factory:)
       @resource = resource
       @resource_factory = resource_factory
     end
 
+    # Converts the Valkyrie Resource to the Solr Document
+    # @note this modifies the Solr Document for the conversion
+    # @return [Hash] the Solr Document for the Valkyrie Resource
     def convert!
+      # Appends the resource type to the Solr Document
       to_h.merge(Valkyrie::Persistence::Solr::Queries::MODEL.to_sym => [resource.internal_resource])
-          .merge(indexer_solr(resource))
+      to_h.merge(indexer_solr(resource))
```

Comment:
Is this actually equivalent? The previous code worked on the Hash returned by the first `merge` call, but doesn't this updated code work on the original?